biz.aQute.bndlib/src/aQute/bnd/component/ComponentDef.java

```diff
@@ -254,8 +254,9 @@ Tag getTag() {
 		return "$".equals(v) ? name : v;
 	}
 
-	void updateVersion(Version version) {
+	void updateVersion(Version version, String reason) {
 		this.version = max(this.version, version);
+		this.versionReason = reason;
```

Comment:
This is wrong since it always changes the reason string even when the supplied version is less than the old version!  The reason should only be changed if the supplied version is greater than the old version.